include/ScrollHelpers.h

```diff
@@ -0,0 +1,81 @@
+/*
+ * ScrollHelpers.h - helper functions for wheel events
+ *
+ * Copyright (c) 2023 Alex <allejok96/gmail>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#ifndef SCROLL_HELPERS_H
+#define SCROLL_HELPERS_H
+
+#include <Qt>
+
+#include "lmms_export.h"
+
+class QWheelEvent;
+
+
+namespace lmms {
+
+
+/*! \brief Mark event as ignored and return true if it matches the orientation
+ *
+ *  Convenience function that may be used for early return in wheelEvent.
+ *  Events that doesn't match the orientation are marked accepted.
+ */
+bool LMMS_EXPORT ignoreScroll(const Qt::Orientation orientation, QWheelEvent* event);
+
+
+
+/*! \brief Return number of scrolled steps.
+ *
+ *  By default it counts standard scroll wheel steps of 15°.
+ *
+ *  If you intend to round or divide WheelEvent::angleDelta() this function should ALWAYS be used to get proper
+ *  support for smooth scrolling mice and trackpads.
+ *
+ *  Only call this function ONCE per event and orientation. Never call it if the event will be ignored.
+ *
+ *  \param event - QWheelEvent to get delta value from.
+ *  \param orientation - Vertical or horizontal.
+ *  \param factor - Scroll speed/precision. If factor=2 it returns 2 for a complete step and 1 for a halfstep.
+ *  \param allowNatural - Whether macOS-style natural scroll should be allowed or inverted to regular scroll.
+ */
+int LMMS_EXPORT getScroll(const QWheelEvent* event, const Qt::Orientation orientation, const float factor,
+	const bool allowNatural);
+
+
+/*! \brief Overload of getScroll
+ *
+ * Returns a positive value if the top of the wheel is moved to the left
+ */
+int LMMS_EXPORT horizontalScroll(const QWheelEvent* event, const float factor = 1, const bool allowNatural = true);
```

Comment:
@tresf changing to `allowNatural = false` in these two places _should_ fix the issue of natural scrolling on knobs. I'll get around to fixing it later...

Reply:
Wouldn't this change belong in [`Knob.cpp`](https://github.com/LMMS/lmms/pull/6700/files#diff-e565ae91eb57594c301bc1e85c6a519df0921e77fa5d81a7fcd692bcea7815d7) instead?  Anyway,  I tried it and it has no effect.

Possibly related: https://github.com/LMMS/lmms/issues/5507#issuecomment-749295142

Reply:
I think I got it... 

```diff
         model()->incValue(verticalScroll(we, scrollFactor) - horizontalScroll(we, scrollFactor));
 
 
+        printf("Inverted: %d\n", we->inverted());
 
         s_textFloat->setText( displayValue() );
```

Shows:

```
Inverted: 1
Inverted: 1
Inverted: 1
Inverted: 1
Inverted: 1
Inverted: 1
Inverted: 1
Inverted: 1
```